pennylane/ops/math/pow.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=too-few-public-methods,function-redefined
+
+import pennylane as qml
+
+
+class Pow(qml.operation.Operator):
+    """Arithmetic operator class representing the power of an operator."""
+
+    def __init__(self, op, exponent, do_queue=True, id=None):
+
+        self.hyperparameters["base"] = op
+        self.hyperparameters["exponent"] = exponent
+
+        super().__init__(*op.parameters, wires=op.wires, do_queue=do_queue, id=id)
+        self._name = f"Pow({op}, {exponent})"
+
+    def __repr__(self):
+        """Constructor-call-like representation."""
+        return f"{self.hyperparameters['base']}**{self.hyperparameters['exponent']}"
+
+    @property
+    def num_wires(self):
+        return len(self.wires)
+
+    @staticmethod
+    def compute_decomposition(*params, wires=None, exponent=None, base=None, **hyperparameters):
+
+        if hasattr(base, "__pow__"):
+            return [base.__pow__(exponent)]
+
+        if exponent == 0:
+            return []
+        if isinstance(exponent, int):
+            if exponent > 0:
+                return [base] * exponent
+            if exponent < 0:
```

Comment:
I think if we have this class, then we don't need additional logic for square root and inverse.

Might be also worth looking at PR #2225  that has all the `__pow__` method shortcuts.  Basically we could have that PR, but with a better default  (`Exp` wrapper class) then raises an error.

Reply:
Sounds good!